core/src/main/scala/org/apache/spark/scheduler/AccumulableInfo.scala

```diff
@@ -34,9 +34,15 @@ class AccumulableInfo private[spark] (
   override def equals(other: Any): Boolean = other match {
     case acc: AccumulableInfo =>
       this.id == acc.id && this.name == acc.name &&
-        this.update == acc.update && this.value == acc.value
+        this.update == acc.update && this.value == acc.value &&
+        this.internal == acc.internal
     case _ => false
   }
+
+  override def hashCode(): Int = {
+    val state = Seq(id, name, update, value, internal)
+    state.map(_.hashCode).reduce(31 * _ + _)
```

Comment:
Ah, something just occurred to me: you've been using `reduce` here but the operation isn't associative or commutative. I think it has to be `reduceLeft` to guarantee the output you want. Maybe update other instances of this in the code (from other PRs) here too?